doc/en/developer/source/policies/community-process.rst

```diff
@@ -12,116 +12,58 @@ The GeoServer community process adheres to the following guidelines to ensure
+  The Project Steering Committee is responsible the strategic direction of the GeoServer project. Central to this responsibility is evaluating proposals and providing a clear decision through a public voting process.
 
 Adding features
 ^^^^^^^^^^^^^^^
 
-During the road map update developers propose new features and improvements to 
-be added to the road map. The following are the prerequisites for proposing a 
+During the development cycle community members propose new features and improvements to 
+be included in the release. The following are the prerequisites for proposing a 
 new feature:
 
 #. The feature has a sponsor. This means either a developer willing to carry out
-   the work or a customer who is paying for it
+   the work or a customer who is paying for it.
 #. The feature has has gone through the :ref:`GSIP <gsip>` process 
    **if necessary**. Whether a feature requires a GSIP is decided by the 
-   community when the feature is proposed
+   community when the feature is proposed.
+
+The determining factor for what release a feature should be included in is based on the estimate of the time to implement the feature, and the current :ref:`release_cycle`. The release cycle includes a "feature freeze" where new features are delayed while stabilize master and cut a new release candidate.
 
-After a feature has met the above prerequisites it is assigned to a release on
-the road map. The determining factor for what release a feature should be 
-assigned to is based on the estimate of the time to implement the feature, and 
-the current :ref:`release_cycle` for the branch the feature is being implemented
-on. If the time to implement the feature does not allow it to be assigned to the
-next release it is scheduled accordingly into a future release.
+New features may be back-ported to the stable series (if technically feasible) after being tried out on master for a month.
 
 .. _release_cycle:
 
 Release cycle
 ^^^^^^^^^^^^^
 
-GeoServer follows a regular release cycle. Usually this cycle is a release 
-every month. However once a development branch has become stable the release
-cycle drops off to every few months. Similarly on an unstable development branch
-the release cycle can be every few months, until the branch becomes stable 
-enough for monthly releases.
\ No newline at end of file
+GeoServer follows a time-boxed release model, this allows new features to be developed for the project with a set expectation of when a feature will be available for use.
+
+The community maintains three active branches:
+
+* master: available for development of new functionality, documentation improvements and bug fixes
+* stable: bug fixes and back-port of new functionality that do not affect the GeoServer API or significantly affect stability.
+* maintenance: bug fixes
+
+GeoServer uses a six month development cycle on the master branch:
+
+  * month 1-5 master is open for development
+  * month 5 a beta release is made on a feature freeze
```

Comment:
Ah, here is why it looked strange, you took a developer perspective, so the series begins when you can develop against it, I always took a user perspective, the series begins when the .0 is released. 

What you are describing in your comment however sounds like going back to the old development model, where we have a month of freeze between the beta and RC, didn't we say it was overkill?

Let me try to reword:
* First day of Month 1: RC of series N-1 is cut, branching it off, master is open for development of the series N
* First day of Month 2: final series N is released, development continues on master for N+1
* Month 3, 4, 5: development of N+1
* First day of month 6: Beta of N+1 is released
* First day of month 7 (Month 1 of N+1): RC of N+1 is released
* First day of month 8 (Month 2 of N+1): final of N+1 is released

And then we repeat. Overall, 7 months of work, with a month of overlap between the stabilization of N-1 and development of N.